redis/src/main/scala/zio/redis/api/Keys.scala

```diff
@@ -344,19 +347,22 @@ trait Keys {
    *   returns an updated cursor that the user needs to use as the cursor argument in the next call along with the
    *   values.
    */
-  final def scan[K: Schema](
+  final def scan(
     cursor: Long,
     pattern: Option[String] = None,
     count: Option[Count] = None,
     `type`: Option[RedisType] = None
-  ): ZIO[RedisExecutor, RedisError, (Long, Chunk[K])] = {
-    val command = RedisCommand(
-      Scan,
-      Tuple4(LongInput, OptionalInput(PatternInput), OptionalInput(CountInput), OptionalInput(RedisTypeInput)),
-      Tuple2Output(ArbitraryOutput[Long](), ChunkOutput(ArbitraryOutput[K]()))
-    )
-    command.run((cursor, pattern.map(Pattern), count, `type`))
-  }
+  ): ResultBuilder1[({ type lambda[x] = (Long, Chunk[x]) })#lambda] =
```

Comment:
Would it make sense to declare a type alias (for example called `Cursor`) instead of using type lambda?

Reply:
We have to use type lambdas here. An alternative is to use the kind-projector, which we discussed [here](https://github.com/mijicd/zio-redis/pull/1#discussion_r777146269).

Reply:
If you're referring to the possibility of defining a "top-level" alias, and using it, while it is possible, I'd say it's not bringing much to the plate as it would be used only here, so I'd rather stick to the current approach.

Reply:
Yes, I was referring to the possibility of defining a top-level alias. Sure, leave it like it is then.